src/lightning/fabric/utilities/testing/_runif.py

```diff
@@ -0,0 +1,151 @@
+# Copyright The Lightning AI team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import operator
+import os
+import sys
+from typing import Any, Optional
+
+import pytest
+import torch
+from lightning_utilities.core.imports import compare_version
+from packaging.version import Version
+
+from lightning.fabric.accelerators import TPUAccelerator
+from lightning.fabric.accelerators.cuda import num_cuda_devices
+from lightning.fabric.accelerators.mps import MPSAccelerator
+from lightning.fabric.strategies.deepspeed import _DEEPSPEED_AVAILABLE
+from lightning.fabric.utilities.imports import _TORCH_GREATER_EQUAL_2_1
+
+
+def _RunIf(
+    *args,
+    min_cuda_gpus: int = 0,
+    min_torch: Optional[str] = None,
+    max_torch: Optional[str] = None,
+    min_python: Optional[str] = None,
+    bf16_cuda: bool = False,
+    tpu: bool = False,
+    mps: Optional[bool] = None,
+    skip_windows: bool = False,
+    standalone: bool = False,
+    deepspeed: bool = False,
+    dynamo: bool = False,
+    **kwargs,
+) -> Any:  # not the real return because it would require that pytest is available
+    """RunIf wrapper for simple marking specific cases, fully compatible with pytest.mark::
+
+    @RunIf(min_torch="0.0")
```

Comment:
Doesn't this need to be under `Example:: `?

Reply:
yes, better be :)